pkg/remote/aws_apigatewayv2_scanner_test.go

```diff
@@ -100,6 +100,87 @@ func TestApiGatewayV2Api(t *testing.T) {
 	}
 }
 
+func TestApiGatewayV2Route(t *testing.T) {
+	dummyError := errors.New("this is an error")
+
+	tests := []struct {
+		test           string
+		mocks          func(*repository.MockApiGatewayV2Repository, *mocks.AlerterInterface)
+		assertExpected func(t *testing.T, got []*resource.Resource)
+		wantErr        error
+	}{
+		{
+			test: "no api gateway v2 api",
+			mocks: func(repository *repository.MockApiGatewayV2Repository, alerter *mocks.AlerterInterface) {
+				repository.On("ListAllApis").Return([]*apigatewayv2.Api{}, nil)
+			},
+			assertExpected: func(t *testing.T, got []*resource.Resource) {
+				assert.Len(t, got, 0)
+			},
+		},
+		{
+			test: "single api gateway v2 api with a single route",
+			mocks: func(repository *repository.MockApiGatewayV2Repository, alerter *mocks.AlerterInterface) {
+				repository.On("ListAllApis").Return([]*apigatewayv2.Api{
+					{ApiId: awssdk.String("a-gateway")},
+				}, nil)
+				repository.On("ListAllApiRoutes", awssdk.String("a-gateway")).
+					Return([]*apigatewayv2.Route{{RouteId: awssdk.String("a-route")}}, nil)
+			},
+			assertExpected: func(t *testing.T, got []*resource.Resource) {
+				assert.Len(t, got, 1)
+
+				assert.Equal(t, "a-route", got[0].ResourceId())
+				assert.Equal(t, resourceaws.AwsApiGatewayV2RouteResourceType, got[0].ResourceType())
+			},
+		},
+		{
+			test: "cannot list api gateway v2 apis",
+			mocks: func(repository *repository.MockApiGatewayV2Repository, alerter *mocks.AlerterInterface) {
+				repository.On("ListAllApis").Return(nil, dummyError)
+				alerter.On("SendAlert", resourceaws.AwsApiGatewayV2RouteResourceType, alerts.NewRemoteAccessDeniedAlert(common.RemoteAWSTerraform, remoteerr.NewResourceListingErrorWithType(dummyError, resourceaws.AwsApiGatewayV2ApiResourceType, resourceaws.AwsApiGatewayV2RouteResourceType), alerts.EnumerationPhase)).Return()
+			},
+			wantErr: remoteerr.NewResourceListingError(dummyError, resourceaws.AwsApiGatewayV2RouteResourceType),
+		},
```

Comment:
You would need to change this test since the returned error will change. Can you add also one test that will return an error if you cannot list API gateway v2 routes